jerry-port/win/jerry-port-win-fs.c

```diff
@@ -0,0 +1,106 @@
+
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "jerryscript-port.h"
+
+#if defined(_WIN32)
+
+#include <stdlib.h>
+#include <string.h>
+
+/**
+ * Normalize a file path.
+ *
+ * @return a newly allocated buffer with the normalized path if the operation is successful,
+ *         NULL otherwise
+ */
+jerry_char_t *
+jerry_port_path_normalize (const jerry_char_t *rel_path_p, /**< path to the referenced module */
+                           jerry_size_t rel_path_size, /**< size of the path */
+                           const jerry_char_t *base_path_p, /**< base path */
+                           jerry_size_t base_path_size) /**< length of the base path */
+{
+  char *path_p = malloc (base_path_size + rel_path_size + 1);
```

Comment:
There is shared part of jerry_port_path_normalize of these functions
Maybe it's better split jerry_path_join, and then
jerry_port_path_real
or
jerry_port_path_full

Reply:
Hi, I am suggesting `jerry_path_join`, not `jerry_port_path_join`,
I am suggesting to split jerry_port_path_normalize into
`jerry_path_normalize`, `jerry_path_join` and `jerry_port_path_full`.
So when porting jerry, we only need implement `jerry_port_path_full`.

Reply:
I see, so essentially what you mean is do the path concatenation in the core library, and then the port function should only need to canonicalize a single input path.

Reply:
> I see, so essentially what you mean is do the path concatenation in the core library, and then the port function should only need to canonicalize a single input path.

Exactly!